js/interface.js

```diff
@@ -307,32 +311,121 @@
     return Fliplet.API.request({
       url: [
         'v1/widgets?include_instances=true&tags=type:menu',
+        '&include_all_versions=true',
         '&appId=' + Fliplet.Env.get('appId'),
-        '&organizationId=' + (Fliplet.Env.get('organizationId') || '')
+        '&organizationId=' + (organizationId || '')
       ].join('')
     }).then(function(response) {
       return Promise.resolve(response.widgets);
     });
   }
 
+  /**
+   * Function to filter organizationMenus by version.
+   * @param {Array} data - array of organizationMenus
+   * @param {Object} latestVersionMenu - current active menu
+   * @returns {Object} latestVersionMenu - object which contains the latest version of current menu package type
+   */
+
+  function sortByVersion(data, latestVersionMenu) {
+    var previousVersion;
+
+    data.forEach(function(menu) {
+      // Compare each record with the previous one by version to find the newest version.
+      if (previousVersion && previousVersion.version !== menu.version) {
+        latestVersionMenu = (previousVersion.version > menu.version) ? previousVersion : menu;
+
+        return;
+      }
+
+      previousVersion = menu;
+
+      // Always write the system menu first, then if iteration finds the newest version, latestVersionMenu will be overwritten.
+      if (!latestVersionMenu) {
+        latestVersionMenu = menu;
+      }
+    });
+
+    return latestVersionMenu;
+  }
+
+  /**
+   * Creates a list of menu widgets to be displayed based on versions and current usage
+   * @param {Array} menus - List of menu widgets
+   * @returns {Array} List of menu widgets to be displayed based on versions and current usage
+   */
+
+  function generateMenuList(menus) {
+    var result = [];
+    var hasActiveMenu = false;
+
+    // System menus don't have organizationId
+    var systemMenus = menus.filter(function(item) {
+      return !item.organizationId;
+    });
+
+    // Cycle iterating the array of system menus to fulfill the necessary conditions.
+    systemMenus.forEach(function(item) {
+      var latestVersionMenu;
+
+      // Creating an array of one type of menu.
+      var packageTypeMenus = menus.filter(function(menu) {
+        if (menu.package) {
```

Comment:
@AndrRyaz I mentioned [in this comment](https://github.com/Fliplet/fliplet-widget-menu-manager/pull/40#discussion_r554036567) that _menus of the same type should have matching packages_.

That means if you want to find all the versions of the same menu, you should be filtering like this:

```suggestion
        return menu.package === item.package;
```

Can you explain why you think this can't work or why it needs to be different from this logic?

Reply:
Of course.
The logic that you suggested strictly compares the menu packages for their coincidence, for example: `com.fliplet.menu.bottom-bar` and `com.fliplet.menu.bottom-bar-andrey` the check will not be correct but these are menus of the same type that should be sorted by version and the menu is displayed which has the latest version, as I understand it. Therefore, I check for a match the split string of its last element because `bottom-bar-andrey` includes 'bottom-bar' so thats mean that this 2 menus have the same type
![изображение](https://user-images.githubusercontent.com/71251809/104740016-19654700-5750-11eb-97d2-33256c6e9712.png)
![изображение](https://user-images.githubusercontent.com/71251809/104740430-9f818d80-5750-11eb-8ce4-841dc354ff97.png)

Reply:
@AndrRyaz I don't think you're quite understanding what I'm saying then (for many times now).

> Menus of the same type should have matching packages

In your example, `com.fliplet.menu.bottom-bar` and `com.fliplet.menu.bottom-bar-andrey` are considered different menus because they have different package names. I know Upplabs creates a custom package during development, but logically that does not mean that to the system they should be considered as the same menu.

If a customer of ours changes `com.fliplet.menu` to `com.fliplet.menu-2`, we need to treat them as different menus.

Reply:
@tonytlwu, I understood, according to the scenario, I thought the opposite, sorry
Fixed